R/ggs_volcano.R

```diff
@@ -0,0 +1,181 @@
+#' enrichment analysis. See more in the main function, [GeneTonic()], to check the
+#' formatting requirements (a minimal set of columns should be present).
+#' @param annotation_obj A `data.frame` object with the feature annotation
+#' information, with at least two columns, `gene_id` and `gene_name`.
+#' @param gtl A `GeneTonic`-list object, containing in its slots the arguments
+#' specified above: `dds`, `res_de`, `res_enrich`, and `annotation_obj` - the names
+#' of the list _must_ be specified following the content they are expecting
+#' @param geneset_id Character specifying the gene set identifier to be plotted
+#' @param genelist A vector of character strings, specifying the identifiers
+#' contained in the row names of the `se` input object.
+#' @param FDR Numeric value, specifying the significance level for thresholding
+#' adjusted p-values. Defaults to 0.05.
+#' @param color Character string to specify color of filtered points in the plot.
+#'  Default color is #1a81c2.
+#' @param volcano_labels Integer, maximum number of labels for the gene sets to be
+#' plotted as labels on the volcano scatter plot. Default value is 25.
+#' @param plot_title Character string, to specify the title of the plot,
+#' displayed over the heatmap. If left to `NULL` as by default, it tries to use
+#' the information on the geneset identifier provided
+#'
+#' @return A plot returned by the [ggplot()] function
+#' @export
+#'
+#' @examples
+#' library("macrophage")
+#' library("DESeq2")
+#' library("org.Hs.eg.db")
+#' library("AnnotationDbi")
+#'
+#' # dds object
+#' data("gse", package = "macrophage")
+#' dds_macrophage <- DESeqDataSet(gse, design = ~line + condition)
+#' rownames(dds_macrophage) <- substr(rownames(dds_macrophage), 1, 15)
+#' dds_macrophage <- estimateSizeFactors(dds_macrophage)
+#'
+#'
+#' # annotation object
+#' anno_df <- data.frame(
+#'   gene_id = rownames(dds_macrophage),
+#'   gene_name = mapIds(org.Hs.eg.db,
+#'                      keys = rownames(dds_macrophage),
+#'                      column = "SYMBOL",
+#'                      keytype = "ENSEMBL"),
+#'   stringsAsFactors = FALSE,
+#'   row.names = rownames(dds_macrophage)
+#' )
+#'
+#' # res object
+#' data(res_de_macrophage, package = "GeneTonic")
+#' res_de <- res_macrophage_IFNg_vs_naive
+#'
+#' # res_enrich object
+#' data(res_enrich_macrophage, package = "GeneTonic")
+#' res_enrich <- shake_topGOtableResult(topgoDE_macrophage_IFNg_vs_naive)
+#' res_enrich <- get_aggrscores(res_enrich, res_de, anno_df)
+#'
+#' ggs_volcano(res_de,
+#'            res_enrich,
+#'            anno_df,
+#'            geneset_id = res_enrich$gs_id[1]
+#'            )
+#'            
+#' alternatively
+#'
+#' # genelist <- sapply(anno_df$gene_name, function(x) str_contains(x, "CD"))
+#' 
+#' # ggs_volcano(res_de,
+#' #             res_enrich,
+#' #             anno_df,
+#' #             genelist = genelist
+#' #             )
```

Comment:
I'd make the call here more explicit, no problem in "showing that we just take a small bunch of genes on ourselves".
Plus: let the code be run, it would be an extra second or so, but no need to worry for that - and if we pass the genes directly, we avoid having to use the `stringr` package 😬 
You could just call the list `cdgenes_list` as in here, or you could use the `chemokines_list`, whatever suits you best (just trying to avoid naming the variable as the parameter itself)